docs/ios/6x/best-practices/common-patterns.md

```diff
@@ -45,32 +46,38 @@ class CheckoutCoordinator {
 
     func startCheckout() {
         // Start a parent span for the whole checkout flow
-        checkoutSpan = Embrace.client.startSpan(name: "checkout_flow")
-        checkoutSpan?.setAttribute(key: "cart_value", value: cart.totalValue)
-        checkoutSpan?.setAttribute(key: "items_count", value: cart.items.count)
+        checkoutSpan = Embrace.client?.buildSpan(name: "checkout_flow").startSpan()
+        checkoutSpan?.setAttribute(key: "cart_value", value: String(cart.totalValue))
+        checkoutSpan?.setAttribute(key: "items_count", value: String(cart.items.count))
 
         navigateToShippingScreen()
     }
 
     func navigateToShippingScreen() {
         // Child span for the shipping step
-        let shippingSpan = Embrace.client.startSpan(name: "checkout_shipping", parent: checkoutSpan)
+        guard let parentSpan = checkoutSpan else { return }
+        let shippingSpan = Embrace.client?.buildSpan(name: "checkout_shipping")
+            .setParent(parentSpan)
+            .startSpan()
         // Show shipping screen
         // ...
-        shippingSpan.end()
+        shippingSpan?.end()
     }
 
     func navigateToPaymentScreen() {
         // Child span for the payment step
-        let paymentSpan = Embrace.client.startSpan(name: "checkout_payment", parent: checkoutSpan)
+        guard let parentSpan = checkoutSpan else { return }
```

Comment:
might `if let` here since we wouldn't want customers to return and cut off the actual functionality